airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -409,4 +434,19 @@ private JobInfoRead readJobFromResult(final ManualOperationResult manualOperatio
     return jobConverter.getJobInfoRead(job);
   }
 
+  private boolean containsBreakingChange(final CatalogDiff diff) {
+    AtomicBoolean isBreaking = new AtomicBoolean(false);
+    diff.getTransforms().stream().forEach(streamTransform -> {
```

Comment:
Nit, this looks like more like a reduce operation rather than a map. In Java it would be written like
```
.collect(false,
(acc, value) -> acc = acc || value.getBreaking(),
(left, right) -> left = left || right
)
```

I find it more explicit about what this is doing but I don't know what is the opinion of the rest of the team.

Reply:
I find this more confusing

Reply:
Not sure if that's better...
```
diff.getTransforms().stream()
    .filter(streamTransform -> streamTransform.getTransformType() == TransformTypeEnum.UPDATE_STREAM)
    .flatMap(streamTransform -> streamTransform.getUpdateStream().stream().map(fieldTransform::getBreaking))
    .anyMatch(Boolean::booleanValue)
```

Reply:
@benmoriceau why do we want to avoid that?

Reply:
I generally think of stream notation as a more functional approach, in that paradigm having a mutable states/side effects would generally be something to avoid.

Reply:
What jimmy is saying. Ideally when using this we need to be able to take the anonymous function in the foreach call and move it to its own method which is not the case here. I would personally prefer using a for loop like `for (A a: as)` if we are not stateless in the anonymous functions.

Reply:
Synced up on Zoom about that